### DISCUSSION TITLE
A question about the particle open boundary

### DISCUSSION THREAD

**Discussion by smallrainbow** [SmileiPIC/Smilei#774] [Q&A]
Dear developers,

**Introduction**
Here are two figures attached. In brief, this is a magnetic reconnection test with open boundaries in the x-direction. 

**Question**
<img width="1178" alt="fig1" src="https://github.com/user-attachments/assets/438b06de-70d5-4e7f-84fc-8b225ff1d394">
In Figure 1, Silver-Müller and remove boundaries are employed for the fields and particles, respectively. It can be seen that at the left and right boundaries, without the injector-like supplementing of the VDF, particles will continuously leak out.

<img width="1166" alt="fig2" src="https://github.com/user-attachments/assets/039bd7ee-53f5-4bb1-bbff-8751dd751d03">
In Figure 2, we applied the thermalize boundary condition for the particles, keeping other conditions unchanged. The results show that the performance in regions farther from Y=0 is better than in Figure 1. However, near Y=0, the particle density at the Harris current sheet is higher than that in the surrounding area. Therefore, simply using a constant density and temperature for the thermalize boundary is not sufficient to fully replenish the particles.

**Request**
As can be seen in both Figure 1 and Figure 2, the open boundary for the fields is already perfect and should be considered one of the best open boundaries in the world. This issue is only concerned with the boundary condition for the particles.

Is it possible to have a boundary condition that copies and pastes particles from the boundary inward by n grid points (n=2, 3, ...) to the i=1 boundary grid? It seems that the open boundary in iPic3D uses this setting to achieve a thermalize-like boundary, allowing properties like density and moments to vary with internal changes. Alternatively, statistical moment resampling could be used, similar to the approach in vpic, although this might slightly affect computational efficiency (especially with a large number of particles).

Thank you very much!

Best regards & have a nice day.
Zhongwei
email: yangz@latmos.ipsl.fr

**Comment by mccoys** [comment]
The easiest way might be to provide the boundary temperature as a profile. But to be honest, I don't see how anyone in the dev team could have time for this in the near future.

As a workaround, if you know basic c++, you could hack your way by modifying the source code for your specific case, and have the boundary temperature vary depending on each particle's y. We can help showing you where that should be put in the code

**Comment by mccoys** [comment]
Actually, making a local measurement of temperature would probably be relatively easy to do

**Comment by smallrainbow** [comment]
Thanks a lot for your prompt reply. Yes, we can try some tests. I am
reading the injector module which has great potentials. Could you so kind
totell how to get neighbor cell statistical values, e.g., temperature,
density and bulk speed? Merci beaucoup!

mccoys ***@***.***> 于 2024年11月27日周三 16:19写道：

**Reply by mccoys** [reply]
I am confused. Are you trying to modify the injector or the thermalizing boundary?

In any case, to start your modifications I suggest you consider so all the particles available (all particles in the given patch) and select them depending on their position.
